From 85c45d153b901d3f69dd5713924039c011477612 Mon Sep 17 00:00:00 2001
From: Daniel Stenberg <daniel@haxx.se>
Date: Wed, 22 Apr 2015 13:58:10 +0200
Subject: [PATCH 2/2] connectionexists: follow-up to fd9d3a1ef1f

PROTOPT_CREDSPERREQUEST still needs to be checked even when NTLM is not
enabled.

Mistake-caught-by: Kamil Dudka
Signed-off-by: Gustavo Zacarias <gustavo@zacarias.com.ar>
---
 lib/url.c | 11 +++++++----
 1 file changed, 7 insertions(+), 4 deletions(-)

diff --git a/lib/url.c b/lib/url.c
index 93f15f1..7dc5c45 100644
--- a/lib/url.c
+++ b/lib/url.c
@@ -3210,9 +3210,11 @@ ConnectionExists(struct SessionHandle *data,
           continue;
       }
 
-#if defined(USE_NTLM)
-      if((!(needle->handler->flags & PROTOPT_CREDSPERREQUEST)) ||
-         (wantNTLMhttp || check->ntlm.state != NTLMSTATE_NONE)) {
+      if((!(needle->handler->flags & PROTOPT_CREDSPERREQUEST))
+#ifdef USE_NTLM
+         || (wantNTLMhttp || check->ntlm.state != NTLMSTATE_NONE)
+#endif
+        ) {
         /* This protocol requires credentials per connection or is HTTP+NTLM,
            so verify that we're using the same name and password as well */
         if(!strequal(needle->user, check->user) ||
@@ -3220,9 +3222,10 @@ ConnectionExists(struct SessionHandle *data,
           /* one of them was different */
           continue;
         }
+#if defined(USE_NTLM)
         credentialsMatch = TRUE;
-      }
 #endif
+      }
 
       if(!needle->bits.httpproxy || needle->handler->flags&PROTOPT_SSL ||
          (needle->bits.httpproxy && check->bits.httpproxy &&
-- 
2.0.5

